servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/single/SingleToCompletionStageTest.java

```diff
@@ -1032,27 +1032,29 @@ private static void verifyInJdkForkJoinThread() {
         if (!currentThread.getName().startsWith(JDK_FORK_JOIN_THREAD_NAME_PREFIX) &&
             // CompletableFuture may use a ThreadPerTaskExecutor executor if available processors is low (e.g. 1).
             !currentThread.getName().startsWith(COMPLETABLE_FUTURE_THREAD_PER_TASK_NAME_PREFIX)) {
-            throw new IllegalStateException("unexpected thread: " + currentThread());
+            throw new AssertionError("unexpected thread: " + currentThread());
         }
     }
 
     private static void verifyInJUnitThread() {
-        if (!currentThread().getName().startsWith(JUNIT_THREAD_PREFIX)) {
-            throw new IllegalStateException("unexpected thread: " + currentThread());
+        if (!currentThread().getName().startsWith(JUNIT_THREAD_PREFIX) &&
+                !currentThread().getName().startsWith(JDK_FORK_JOIN_THREAD_NAME_PREFIX)) {
+            throw new AssertionError("unexpected thread: " + currentThread());
         }
     }
 
     private static void verifyInStOrJdkThread() {
         if (!currentThread().getName().startsWith(ST_THREAD_PREFIX_NAME) &&
                 !currentThread().getName().startsWith(JDK_THREAD_NAME_PREFIX)) {
-            throw new IllegalStateException("unexpected thread: " + currentThread());
+            throw new AssertionError("unexpected thread: " + currentThread());
         }
     }
 
     private static void verifyInStOrJUnitThread() {
         if (!currentThread().getName().startsWith(ST_THREAD_PREFIX_NAME) &&
-                !currentThread().getName().startsWith(JUNIT_THREAD_PREFIX)) {
-            throw new IllegalStateException("unexpected thread: " + currentThread());
+                !currentThread().getName().startsWith(JUNIT_THREAD_PREFIX) &&
+                !currentThread().getName().startsWith(JDK_FORK_JOIN_THREAD_NAME_PREFIX)) {
+            throw new AssertionError("unexpected thread: " + currentThread());
```

Comment:
If those are improvements for tests, consider opening a separate PR.

Reply:
The test parallelism causes JUnit to run the tests on the default ForkJoinPool so the thread observed changes from `Test-` to `ForkJoinPool-`